test/tsconfig.test.json

```diff
@@ -3,7 +3,14 @@
   "compilerOptions": {
     "emitDeclarationOnly": false,
     "noEmit": true,
-    "allowImportingTsExtensions": true
+    "allowImportingTsExtensions": true,
+    // TODO: This setting is not compatible with the default definition of
+    // DocumentTypeDecoration from '@graphql-codegen-types/core'. That
+    // definition includes `{__apiType?: ...}` as an optional, so assigning it
+    // to an explicit `undefined` is impossible under
+    // exactOptionalPropertyTypes. We only need to work around this in tests,
+    // since that is where we use concrete examples of DocumentTypeDecoration.
+    "exactOptionalPropertyTypes": false
```

Comment:
A bit cursed, but I do not know of a way to work around this from the call-site. [Here is relevant part in graphql-typed-document-node](https://github.com/dotansimha/graphql-typed-document-node/blob/925c2b0483ae602eec38253acf602c2aefe383ec/packages/core/src/index.ts#L3-L10)